components/raftstore/src/store/peer.rs

```diff
@@ -2275,6 +2325,7 @@ where
                             "region_id" => self.region_id,
                         );
                     }
+                    self.maybe_witness_transfer_leader(ctx);
```

Comment:
I have a question:

As the witness is forbidden to read and write, and if it is a witness, the transfer leader will be executed immediately. So the call to this function can be brought to the front of `StateRole::Leader => {}` to bypass renew leader lease?

Reply:
Nice catch, it should be checked in `should_renew_lease`.

Reply:
On second thought, it's unnecessary. It rejects' witness's read in the local reader.